scripts/pre_commit_linter.py

```diff
@@ -1927,6 +1934,112 @@ def _check_for_copyright_notice(self):
+                            print ('%s --> Pattern on line %s is invalid. Use '
+                                   'full path relative to the root directory'
+                                   % (codeowner_file, line_num))
+                            failed = True
+                        # The double asterisk pattern is supported by the
+                        # CODEOWNERS syntax but not the glob in Python 2.
+                        # The following condition checks this.
+                        if '**' in line_in_concern:
+                            print ('%s --> Pattern on line %s is invalid. '
+                                   '\'**\' wildcard not allowed' % (
+                                        codeowner_file, line_num))
+                            failed = True
+                        path_patterns.append(line_in_concern.replace(
+                            '/', '', 1))
+
+            # Checks that every dir/file is covered under CODEOWNERS.
+            for root, _, filename in os.walk('.'):
+                for file_name in filename:
+                    # This bool checks if the file belongs to the root
+                    # oppia directory.
+                    is_root_file = False
+                    if os.path.join(root, file_name) in CODEOWNER_FILE_PATHS:
+                        is_root_file = True
+                    if (any(root.startswith(
+                            dir_path) for dir_path in CODEOWNER_DIR_PATHS) or
+                                is_root_file):
+                        match = False
+                        # Ignore .pyc and __init__.py files.
+                        if file_name.endswith(
+                                '.pyc') or file_name == '__init__.py':
+                            match = True
+                            continue
+                        for path_to_match in path_patterns:
+                            if path_to_match.split('/')[-1]:
+                                level = len(path_to_match.split('/'))
+                            else:
+                                level = len(path_to_match.split('/')) - 1
+                            if os.path.join(*((
+                                    os.path.join(root, file_name).replace(
+                                        './', '', 1)).split(
+                                            '/')[:level])) in glob.glob(
+                                                path_to_match):
+                                match = True
+                                break
+                        if not match and self.verbose_mode_enabled:
+                            print ('WARNING! %s/%s is not covered under '
+                                   'CODEOWNERS' % (root, file_name))
+
+            if failed:
+                summary_message = '%s   CODEOWNERS file check failed' % (
+                    _MESSAGE_TYPE_FAILED)
+                print summary_message
```

Comment:
Pull out this and the below statement. They are being repeated in both the blocks.

Reply:
Done.